metricflow/execution/dataflow_to_execution.py

```diff
@@ -71,15 +73,17 @@ def __init__(
         self._sql_plan_renderer = sql_plan_renderer
         self._sql_client = sql_client
         self._optimization_level = sql_optimization_level
+        self._spec_output_order: Sequence[InstanceSpec] = ()
```

Comment:
Same here - seems like this class should be request-agnostic. But if this was a property on the sink node, it wouldn't be needed here.